In [1]:
import pandas as pd
import plotly.express as px

# Load the dataset
df = pd.read_csv('/Users/manishghoshal/Downloads/Futurense_Internship-main/Final Internship Capstone Project/Financial Analytics/financialanalytics.csv')


# 1. Calculate average Earnings per Share (EPS) for each industry
eps_by_industry = df.groupby('Industry')['EarningsPerShare'].mean().reset_index()

# Print the average EPS for each industry
print("Average EPS by Industry:")
print(eps_by_industry)

Average EPS by Industry:
     Industry  EarningsPerShare
0     Finance         -0.153240
1  Healthcare         -0.233913
2  Technology          0.084855


In [2]:

# 2. Find the industry with the highest EPS
highest_eps_industry = eps_by_industry.loc[eps_by_industry['EarningsPerShare'].idxmax()]

# 3. Calculate the mean EPS across all industries
mean_eps = df['EarningsPerShare'].mean()

# Print the mean EPS
print("\nMean EPS across all industries:", mean_eps)

# Print the industry with the highest EPS
print("Industry with the highest EPS:", highest_eps_industry['Industry'])


Mean EPS across all industries: -0.1011296125895762
Industry with the highest EPS: Technology


In [3]:
# Calculate the range of EPS and assess the significant difference
eps_range = eps_by_industry['EarningsPerShare'].max() - eps_by_industry['EarningsPerShare'].min()
significant_difference = eps_range > 0.1  # Adjust the threshold as needed

# Print if there is a significant difference in EPS among different industries
print("Is there a significant difference in EPS among different industries?", significant_difference)


Is there a significant difference in EPS among different industries? True


In [4]:
# Plotting EPS by industry with different colors
fig = px.bar(eps_by_industry, x='Industry', y='EarningsPerShare', title='Average Earnings per Share (EPS) by Industry',
             color='Industry')
fig.update_traces(opacity=0.8)
fig.update_layout(
    xaxis_title="Industry",
    yaxis_title="Average EPS",
    showlegend=False,
)
fig.add_annotation(
    x=highest_eps_industry['Industry'],
    y=highest_eps_industry['EarningsPerShare'],
    text=f"Highest EPS: {highest_eps_industry['EarningsPerShare']:.2f}",
    showarrow=True,
    arrowhead=1,
)
fig.show()

**Infer findings:**

1. **Industry with Highest EPS:**
   - The industry 'Technology' has the highest average EPS of 0.0849.

2. **EPS Comparison Among Different Industries:**
   - The bar plot shows the average EPS for each industry, allowing comparison of EPS performance.

3.**Calculated mean EPS across all industries:**
   - Mean EPS across all industries: -0.1011296125895762

4.**Average EPS by Industry:**

     Industry  EarningsPerShare
0     Finance         -0.153240

1  Healthcare         -0.233913

2  Technology          0.084855
